iphone/Classes/TiUIListView.m

```diff
@@ -2013,13 +2012,9 @@ - (void)scrollViewDidScroll:(UIScrollView *)scrollView
           [eventArgs setValue:NUMINTEGER([indexPath section]) forKey:@"firstVisibleSectionIndex"];
           [eventArgs setValue:section forKey:@"firstVisibleSection"];
           [eventArgs setValue:[section itemAtIndex:[indexPath row]] forKey:@"firstVisibleItem"];
+          [eventArgs setValue:NUMINTEGER(topSpacing) forKey:@"top"];
 
-          if (lastVisibleItem != [indexPath row] || lastVisibleSection != [indexPath section]) {
-            // only log if the item changes
-            [self.proxy fireEvent:@"scrolling" withObject:eventArgs propagate:NO];
-            lastVisibleItem = [indexPath row];
-            lastVisibleSection = [indexPath section];
-          }
+          [self.proxy fireEvent:@"scrolling" withObject:eventArgs propagate:NO];
```

Comment:
I am concerned about the performance impact of this change, as it will now always fire and not only if the item changed. The reason the check was added initially was also performance-related, so this could affect lists even if not using the event.

Reply:
Web browsers fire the `onscroll` event constantly, so it was my suggestion to do the same and only always fire the event as scrolling. Is there a way to only fire the event if there are listeners?

Reply:
The web browser does not have the kroll proxy behind it that blocks the main thread every time :) It's a really critical piece of code here.

Reply:
I've added the `only change when a new item` code so it won't fire the event all the time without new info since there wasn't a y-coordinate in the response before. It was just unnecessary to see it all the time when there was no change.  I didn't do that for performance reasons BUT I didn't do performance analysis on iOS :smile: 

On Android I didn't see any impact in CPU usage with or without this PR. 
But of course it won't be useful if it blocks the main thread for other parts in your app!

I'm happy to add it behind a flag again so you can enable/disable it and it will only fire when you need the value to constantly change:
* default is still normal scroll behavior (firing only when you change direction) - current default in 12.1.1.GA
* you can enable `continuousUpdate` and that will fire every time a new item is at the top (also already in 12.1.1GA)
* add another flag `forceUpdates` that will fire all the time (new way) - with a hint in the docs that it should only be used with caution and when you have a reason for it e.g. moving out elements depending on your scroll depth.

Reply:
Yep, a flag may actually be good, at least until it's wider adopted. Are there specific community requests or a personal dependency on your end to add it?

Reply:
community request for the `top` value to build something like this http://jsfiddle.net/mariusc23/s6mLJ/31/ in Titanium. And it's only useful if the `top` is reported all the time and not only when the item changes (which is the current workaround)

Reply:
back to the original PR: `forceUpdates` parameter is back :smile: